chronos/config.nim

```diff
@@ -0,0 +1,50 @@
+## Compile-time configuration options for chronos that control the availability
+## of various strictness and debuggability options. In general, debug helpers
+## are enabled when `debug` is defined while strictness options are introduced
+## in transition periods leading up to a breaking release that starts enforcing
+## them and removes the option.
+##
+## `chronosv4` is a preview flag to enable v4 semantics - in particular, it
+## enables strict exception checking and disables parts of the deprecated API
+when (NimMajor, NimMinor) >= (1, 4):
+  const
+    chronosStrictException* {.booldefine.}: bool = defined(chronosv4)
+      ## Require that `async` code raises only derivatives of `CatchableError` and
+      ## not `Exception` - forward declarations, methods and `proc` types used
+      ## from within `async` code may need to be be explicitly annotated with
+      ## `raises: [CatchableError]` when this mode is enabled.
+
+    chronosStackTrace* {.booldefine.}: bool = defined(debug)
+      ## Include stack traces in futures for creation and completion points
+
+    chronosFutureId* {.booldefine.}: bool = defined(debug)
+      ## Generate a unique `id` for every future - when disabled, the address of
+      ## the future will be used instead
+
+    chronosFutureTracking* {.booldefine.}: bool = defined(debug)
+      ## Generate a linked list of all relevant in-flight futures to allow
+      ## detection of hung tasks - works best with stack traces enabled
```

Comment:
> - works best with stack traces enabled - totally incorrect statement

It doesn't depend on `stack traces` at all and uses nim's compiler source line information.